### PR TITLE
globalsort: opt merge step mem to run on 8c

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: ref #50752

Problem Summary:

### What changed and how does it work?
- limit max merging file per thread to 250
- limit one writer mem limit to 128Mi
- limit max upload worker in merge-step to 8 per thread
- calculate part-size by kv goup
- add monitor for upload workers
- some refactor
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
run 2 import 1t job with thread=8 and force-merge on a 16c machine, success. and no performance regression in this case
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/executor/importer/precheck.go [hunk 1]
```diff
@@ -55,9 +55,9 @@ func (e *LoadDataController) CheckRequirements(ctx context.Context, conn sqlexec
 		}
 		// run global sort with < 16 thread might OOM on merge step
 		// TODO: remove this limit after control memory usage.
-		if e.IsGlobalSort() && e.ThreadCnt < 16 {
-			return exeerrors.ErrLoadDataPreCheckFailed.FastGenByArgs("global sort requires at least 16 threads")
-		}
+		//if e.IsGlobalSort() && e.ThreadCnt < 8 {
```
Comment (D3Hunter) [inline pkg/executor/importer/precheck.go:58]: need revert

**Comment by D3Hunter** [comment]
the restriction that thread>=16 will change later.

**Review by wjhuang2016 — APPROVED** [review]

File: br/pkg/storage/gcs_extra.go [hunk 2]
```diff
@@ -141,6 +141,7 @@ func (w *GCSWriter) readChunk(ch chan chunk) {
 			break
 		}
 
+		activeUploadWorkerCnt.Add(1)
```
Comment (tangenta) [inline br/pkg/storage/gcs_extra.go:144]: Maybe defer `activeUploadWorkerCnt.Add(-1)` here to handle panic case.

**Review by tangenta — APPROVED** [review]

**Review by 3pointer — APPROVED** [review]

**Pull request merged**